pkg/providers/common/common.go

```diff
@@ -67,17 +69,18 @@ func GenerateSSHAuthKey(writer filewriter.FileWriter) (string, error) {
 		return "", fmt.Errorf("generating ssh key pair: %v", err)
 	}
 
-	logger.Info(fmt.Sprintf(
-		"Private key saved to %[1]s. Use 'ssh -i %[1]s <username>@<Node-IP-Address>' to login to your cluster node",
-		privateKeyPath,
-	))
+	logger.Info("Private key saved", "sshCommand", fmt.Sprintf("ssh -i %s <username>@<Node-IP-Address>", privateKeyPath))
 
 	key := string(sshAuthorizedKeyBytes)
 	key = strings.TrimRight(key, "\n")
 
 	return key, nil
 }
 
+func ValidateSSHAuthKeyExists(writer filewriter.FileWriter) bool {
```

Comment:
I would pass the directory path here instead of the writer. That makes the method more reusable since it doesn't depend on having a file writer